backend/src/main/kotlin/outcobra/server/util/DateRange.kt

```diff
@@ -0,0 +1,27 @@
+package outcobra.server.util
+
+import outcobra.server.exception.ValidationKey
+import java.time.LocalDate
+
+/**
+ * Every instance of this class represents a range of time between two dates
+ * @param from the start date  as a [LocalDate]
+ * @param to the end date as a [LocalDate]
+ * @author Florian Bürgi
+ * @since <since>
+ */
+class DateRange(val from: LocalDate, val to: LocalDate) {
+
+    /**
+     * checks if the range of the first 2 params overlap with the rang of the second 2 params
+     * @return [Boolean] date ranges do overlap or not
+     */
+    infix fun checkOverlap(dateRange: DateRange): Boolean {
```

Comment:
I would change the parameter name to `otherRange` or `toCompare` or something like this but you don't have to